src/main/java/org/jabref/logic/importer/fetcher/GoogleScholar.java

```diff
@@ -64,36 +66,51 @@ public GoogleScholar(ImportFormatPreferences importFormatPreferences) {
         }
 
         try {
+            // title search
             URIBuilder uriBuilder = new URIBuilder(SEARCH_IN_TITLE_URL);
             uriBuilder.addParameter("as_q", "");
+            // as_epq as exact phrase
             uriBuilder.addParameter("as_epq", entry.getField(FieldName.TITLE).orElse(null));
+            // as_occt field to search in
             uriBuilder.addParameter("as_occt", "title");
 
-            Document doc = Jsoup.connect(uriBuilder.toString()).userAgent(URLDownload.USER_AGENT).get();
-            // Check results for PDF link
-            // TODO: link always on first result or none?
-            for (int i = 0; i < NUM_RESULTS; i++) {
-                Elements link = doc.select(String.format("div[data-rp=%S] div.gs_or_ggsm a", i));
-
-                if (link.first() != null) {
-                    String target = link.first().attr("href");
-                    // link present?
-                    if (!"".equals(target) && new URLDownload(target).isPdf()) {
-                        // TODO: check title inside pdf + length?
-                        // TODO: report error function needed?! query -> result
-                        LOGGER.info("Fulltext PDF found @ Google: " + target);
-                        pdfLink = Optional.of(new URL(target));
-                        break;
-                    }
-                }
-            }
+            pdfLink = search(uriBuilder.toString());
         } catch (URISyntaxException e) {
             throw new FetcherException("Building URI failed.", e);
         }
 
         return pdfLink;
     }
 
+    @Override
+    public TrustLevel getTrustLevel() {
+        return TrustLevel.META_SEARCH;
+    }
+
+    private Optional<URL> search(String url) throws IOException {
+        Optional<URL> pdfLink = Optional.empty();
+
+        Document doc = Jsoup.connect(url).userAgent(URLDownload.USER_AGENT).get();
+        // Check results for PDF link
+        // TODO: link always on first result or none?
+        for (int i = 0; i < NUM_RESULTS; i++) {
+            Elements link = doc.select(String.format("div[data-rp=%S] div.gs_or_ggsm a", i));
+
+            if (link.first() != null) {
+                String target = link.first().attr("href");
+                // link present?
+                if (!"".equals(target) && new URLDownload(target).isPdf()) {
+                    // TODO: check title inside pdf + length?
+                    // TODO: report error function needed?! query -> result
+                    LOGGER.info("Fulltext PDF found @ Google: " + target);
```

Comment:
Is this log statement really needed? If you do a lot of downloads it kind of clutters up the log imho...